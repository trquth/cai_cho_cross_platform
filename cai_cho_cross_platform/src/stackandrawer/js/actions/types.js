// league
export const GET_LEAGUE = 'GET_LEAGUE';

// match
export const GET_TEAM = 'GET_TEAM';

// navigation
export const TOGGLE_DRAWER = 'TOGGLE_DRAWER';
export const BACK = 'BACK';
