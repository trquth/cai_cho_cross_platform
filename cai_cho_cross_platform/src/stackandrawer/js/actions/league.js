import { GET_LEAGUE } from './types';
import { getTeams } from '../db/helpers';

export function getLeague(currentLeagueRoute) {
  const teams = getTeams(currentLeagueRoute);

  return {
    type: GET_LEAGUE,
    payload: { teams },
  };
}
