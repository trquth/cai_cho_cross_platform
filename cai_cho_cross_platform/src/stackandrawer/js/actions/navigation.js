import { BACK, TOGGLE_DRAWER } from './types';

export function back() {
  return { type: BACK };
}

export function toggleDrawer() {
  return { type: TOGGLE_DRAWER };
}
