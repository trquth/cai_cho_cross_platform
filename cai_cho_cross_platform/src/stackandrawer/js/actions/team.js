import { GET_TEAM } from './types';
import { getMatches } from '../db/helpers';
import { getTeamLogo } from '../api';

function teamAction(currentLeague, currentTeam, image) {
  const matches = getMatches(currentLeague, currentTeam);
  return { type: GET_TEAM, payload: { matches, currentTeam, image } };
}

export function getTeam(currentLeague, currentTeam) {
  return dispatch => getTeamLogo(currentTeam)
    .then(response => response.json())
    .then(response => dispatch(teamAction(currentLeague, currentTeam, response.Image)))
    .catch(() => dispatch(teamAction(currentLeague, currentTeam, null)))
  ;
}
