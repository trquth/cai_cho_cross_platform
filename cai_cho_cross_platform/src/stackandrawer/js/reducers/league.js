import { GET_LEAGUE } from '../actions/types';
import { getTeams } from '../db/helpers';

const initialState = {
  teams: getTeams('PremierLeague'),
};

export default function (state = initialState, action) {
  if (action.type === GET_LEAGUE) {
    return { ...action.payload };
  }

  return state;
}
