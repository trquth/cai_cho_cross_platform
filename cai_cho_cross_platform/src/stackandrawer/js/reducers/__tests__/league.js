import mockStore from 'redux-mock-store';
import { getLeague } from '../../actions/league';
import league from '../league';

const store = mockStore();

beforeEach(() => {
  store.clearActions();
});

it('INITIAL STATE', () => {
  const action = { type: 'DUMMY' };
  expect(league(undefined, action)).toMatchSnapshot();
});

it('GET_LEAGUE (LaLiga)', () => {
  expect(league(undefined, getLeague('LaLiga'))).toMatchSnapshot();
});
