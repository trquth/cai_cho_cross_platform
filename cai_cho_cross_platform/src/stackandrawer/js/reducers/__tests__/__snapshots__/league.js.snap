// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET_LEAGUE (LaLiga) 1`] = `
Object {
  "teams": Object {
    "clubs": Array [
      Object {
        "code": "BAR",
        "key": "barcelona",
        "name": "Barcelona",
      },
      Object {
        "code": "ESP",
        "key": "espanyol",
        "name": "Espanyol",
      },
      Object {
        "code": "RMD",
        "key": "madrid",
        "name": "R. Madrid",
      },
      Object {
        "code": "ATL",
        "key": "atletico",
        "name": "Atlético",
      },
      Object {
        "code": "SEV",
        "key": "sevilla",
        "name": "Sevilla",
      },
      Object {
        "code": "VAL",
        "key": "valencia",
        "name": "Valencia",
      },
      Object {
        "code": "MAG",
        "key": "malaga",
        "name": "Málaga",
      },
      Object {
        "code": "ATH",
        "key": "athletic",
        "name": "Athletic",
      },
      Object {
        "code": "GRA",
        "key": "granada",
        "name": "Granada",
      },
      Object {
        "code": "CEL",
        "key": "celta",
        "name": "Celta",
      },
      Object {
        "code": "RSO",
        "key": "realsociedad",
        "name": "R. Sociedad",
      },
      Object {
        "code": "DEP",
        "key": "deportivo",
        "name": "Deportivo",
      },
      Object {
        "code": null,
        "key": "eibar",
        "name": "Eibar",
      },
      Object {
        "code": "BET",
        "key": "betis",
        "name": "Betis",
      },
      Object {
        "code": null,
        "key": "sporting",
        "name": "Sporting",
      },
      Object {
        "code": null,
        "key": "laspalmas",
        "name": "Las Palmas",
      },
      Object {
        "code": "OSA",
        "key": "osasuna",
        "name": "Osasuna",
      },
      Object {
        "code": "VLL",
        "key": "villareal",
        "name": "Villarreal CF",
      },
      Object {
        "code": null,
        "key": "alaves",
        "name": "Alavés",
      },
      Object {
        "code": null,
        "key": "leganes",
        "name": "Club Deportivo Leganés",
      },
    ],
    "name": "Primera División 2016/17",
    "routeName": "LaLiga",
  },
}
`;

exports[`INITIAL STATE 1`] = `
Object {
  "teams": Object {
    "clubs": Array [
      Object {
        "code": "CHE",
        "key": "chelsea",
        "name": "Chelsea",
      },
      Object {
        "code": "ARS",
        "key": "arsenal",
        "name": "Arsenal",
      },
      Object {
        "code": "TOT",
        "key": "tottenham",
        "name": "Tottenham Hotspur",
      },
      Object {
        "code": "WHU",
        "key": "westham",
        "name": "West Ham United",
      },
      Object {
        "code": "CRY",
        "key": "crystalpalace",
        "name": "Crystal Palace",
      },
      Object {
        "code": "MUN",
        "key": "manutd",
        "name": "Manchester United",
      },
      Object {
        "code": "MCI",
        "key": "mancity",
        "name": "Manchester City",
      },
      Object {
        "code": "EVE",
        "key": "everton",
        "name": "Everton",
      },
      Object {
        "code": "LIV",
        "key": "liverpool",
        "name": "Liverpool",
      },
      Object {
        "code": "WBA",
        "key": "westbrom",
        "name": "West Bromwich Albion",
      },
      Object {
        "code": "STK",
        "key": "stoke",
        "name": "Stoke City",
      },
      Object {
        "code": "SUN",
        "key": "sunderland",
        "name": "Sunderland",
      },
      Object {
        "code": "SOU",
        "key": "southampton",
        "name": "Southampton",
      },
      Object {
        "code": "LEI",
        "key": "leicester",
        "name": "Leicester City",
      },
      Object {
        "code": "BOU",
        "key": "bournemouth",
        "name": "Bournemouth",
      },
      Object {
        "code": "WAT",
        "key": "watford",
        "name": "Watford",
      },
      Object {
        "code": "BUR",
        "key": "burnley",
        "name": "Burnley",
      },
      Object {
        "code": "HUL",
        "key": "hull",
        "name": "Hull City",
      },
      Object {
        "code": "MFC",
        "key": "middlesbrough",
        "name": "Middlesbrough",
      },
      Object {
        "code": "SWA",
        "key": "swansea",
        "name": "Swansea",
      },
    ],
    "name": "English Premier League 2016/17",
    "routeName": "PremierLeague",
  },
}
`;
